src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -1848,6 +1848,29 @@ def test_fw_literal_node(fortran_writer):
     lit1 = Literal('a', CHARACTER_TYPE)
     result = fortran_writer(lit1)
     assert result == "'a'"
+    # An empty character string is valid
+    lit1 = Literal('', CHARACTER_TYPE)
+    result = fortran_writer(lit1)
+    assert result == "''"
+
+    # Check that we generate the correct quotation marks if the literal itself
```

Comment:
As suggested earlier, I think we shouldn't support '' or "" at the moment and raise an exception if we find them. So this test would need to be simplified and another test added to check for an exception.

Reply:
Discussion continues ...

Reply:
Test updated now I've seen the light.